core/genesis.go

```diff
@@ -190,39 +190,6 @@ func (ga *GenesisAlloc) flush(db ethdb.Database, triedb *trie.Database, blockhas
 	return nil
 }
 
-// CommitGenesisState loads the stored genesis state with the given block
```

Comment:
Note to self: Do we use this?

Reply:
Already drafted a PR in the monorepo that pulls in this geth draft, and we can pass every test without it. So I don't think we use it.

Reply:
Confirmed geth will start up with no genesis state and begin snap sync using these changes.  The header validation fails once it gets back to the pre-bedrock blocks but that's a straight forward fix. It's a huge win to have built-in support for missing states in upstream geth.